The config param of the linter.verify() method doesn't support "extends" property

**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** 6.5.0
* **npm Version:** 3.10.6

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{"extends": "eslint:recommended"}
```

**What did you do? Please include the actual source code causing the issue.**
```js
const linter = require("eslint").linter;
const warnings = linter.verify("console.log();", {"extends": "eslint:recommended"});
```

**What did you expect to happen?**
I expected `linter.verify()` to return an array with two warnings:
- Unexpected console statement (no-console)
- Newline required at end of file but not found (eol-last)

**What actually happened? Please include the actual, raw output from ESLint.**
`linter.verify()` returned `[]` (an empty array)

Also
```js
linter.verify("console.log();", {"extends": "blablablabla"});
```
should throw an error saying `Cannot find module 'eslint-config-blablablabla'`, but instead it just executes successfully and returns an array with the "Undeclared 'console'" warning.

--- a/docs/developer-guide/nodejs-api.md
+++ b/docs/developer-guide/nodejs-api.md
@@ -51,7 +51,7 @@ var codeLines = SourceCode.splitLines(code);
 
 ## linter
 
-The `linter` object does the actual evaluation of the JavaScript code. It doesn't do any filesystem operations, it simply parses and reports on the code. You can retrieve `linter` like this:
+The `linter` object does the actual evaluation of the JavaScript code. It doesn't do any filesystem operations, it simply parses and reports on the code. In particular, the `linter` object does not process configuration objects or files. You can retrieve `linter` like this:
 
 ```js
 var linter = require("eslint").linter;
@@ -60,7 +60,8 @@ var linter = require("eslint").linter;
 The most important method on `linter` is `verify()`, which initiates linting of the given text. This method accepts four arguments:
 
 * `code` - the source code to lint (a string or instance of `SourceCode`).
-* `config` - a configuration object that is equivalent to an eslintrc file.
+* `config` - a configuration object that has been processed and normalized by CLIEngine using eslintrc files and/or other configuration arguments.
+    * **Note**: If you want to lint text and have your configuration be read and processed, use CLIEngine's [`executeOnFiles`](#executeonfiles) or [`executeOnText`](#executeontext) instead.
 * `optionsOrFilename` - (optional) Additional options for this run or a string representing the filename to associate with the code being linted.
     * `filename` - (optional) the filename to associate with the source code.
     * `saveState` - (optional) see below. This will override any value passed as the fourth argument if an options object is used here instead of the filename.

